src/loaders/genome_collection/parse_computed_genome_attribs.py

```diff
@@ -191,6 +234,233 @@ def checkm2(root_dir, kbase_collection, load_ver):
     return checkm2_docs
 
 
+def _process_trait(row: Dict[str, Union[str, float, int, bool]],
+                   traits_meta: Dict[str, Dict[str, str]],
+                   traits_val: Dict[str, List[Dict[str, Union[int, float, bool]]]],
+                   data_id: str, ):
+    # Process a row from the trait file and update the global traits metadata and value lists accordingly
+
+    _append_or_check_trait(traits_meta, row[_SYS_TRAIT_ID], row[_SYS_TRAIT_NAME], row[_SYS_TRAIT_DESCRIPTION],
+                           row[_SYS_TRAIT_CATEGORY])
+    _append_trait_val(traits_meta, traits_val, row[_SYS_TRAIT_ID], row[_SYS_TRAIT_VALUE], data_id)
+
+
+def _append_trait_val(
+        traits_meta: Dict[str, Dict[str, str]],
+        traits_val: Dict[str, List[Dict[str, Union[int, float, bool]]]],
+        trait_id: str,
+        trait_value: float | int | bool,
+        data_id: str):
+    # Append a trait value to the global traits value list
+
+    try:
+        trait_index = traits_meta[trait_id][_SYS_TRAIT_INDEX]
+    except KeyError as e:
+        raise ValueError(f'Unable to find trait ID {trait_id}') from e
+
+    if data_id not in traits_val:
+        traits_val[data_id] = list()
+
+    traits_val[data_id].append({_SYS_TRAIT_INDEX: trait_index,  # used as column index in the heatmap
+                                _SYS_TRAIT_VALUE: trait_value})
+
+
+def _append_or_check_trait(
+        global_traits_meta: Dict[str, Dict[str, str]],
+        trait_id: str,
+        trait_name: str,
+        trait_description: str,
+        trait_category: str):
+    # Append a new trait to the global traits dictionary or check if the trait information is consistent
+
+    if trait_id not in global_traits_meta:
+        # add new trait
+        global_traits_meta[trait_id] = {
+            _SYS_TRAIT_INDEX: len(global_traits_meta),  # used as column index in the heatmap
```

Comment:
Microtrait has as specific trait ordering specified in its output. I'm not sure this approach preserves that ordering

Reply:
updated to use the diplay-order from granularity table as trait/column index.

Reply:
Per our huddle my understanding is the cell processing needs to be updated to do the same thing

Reply:
👍

Reply:
:+1: